tests/smoke/multi_cluster_deploy_test.go

```diff
@@ -0,0 +1,217 @@
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rook/rook/tests/framework/clients"
+	"github.com/rook/rook/tests/framework/enums"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	utilversion "k8s.io/kubernetes/pkg/util/version"
+)
+
+func TestInstallingMultipleRookCluster(t *testing.T) {
+	suite.Run(t, new(MultiRookClusterDeploySuite))
+}
+
+type MultiRookClusterDeploySuite struct {
+	suite.Suite
+	helper1     *clients.TestClient
+	helper2     *clients.TestClient
+	k8sh        *utils.K8sHelper
+	installer   *installer.InstallHelper
+	installData *installer.InstallData
+}
+
+//Deploy Multiple Rook clusters
+func (mrc *MultiRookClusterDeploySuite) SetupSuite() {
+	kh, err := utils.CreateK8sHelper()
+	require.NoError(mrc.T(), err)
+
+	mrc.k8sh = kh
+	mrc.installer = installer.NewK8sRookhelper(kh.Clientset)
+	mrc.installData = installer.NewK8sInstallData()
+
+	err = mrc.installer.CreateK8sRookOperator()
+	require.NoError(mrc.T(), err)
+
+	require.True(mrc.T(), kh.IsPodInExpectedState("rook-operator", "default", "Running"),
+		"Make sure rook-operator is in running state")
+
+	time.Sleep(10 * time.Second)
+
+	err = mrc.installer.CreateK8sRookCluster(clusterNamespace1)
+	require.NoError(mrc.T(), err)
+
+	err = mrc.installer.CreateK8sRookToolbox(clusterNamespace1)
+	require.NoError(mrc.T(), err)
+
+	err = mrc.installer.CreateK8sRookCluster(clusterNamespace2)
+	require.NoError(mrc.T(), err)
+
+	err = mrc.installer.CreateK8sRookToolbox(clusterNamespace2)
+	require.NoError(mrc.T(), err)
+
+	mrc.helper1, err = clients.CreateTestClient(enums.Kubernetes, kh, clusterNamespace1)
+	require.Nil(mrc.T(), err)
+
+	mrc.helper2, err = clients.CreateTestClient(enums.Kubernetes, kh, clusterNamespace2)
+	require.Nil(mrc.T(), err)
+
+}
+
+func (mrc *MultiRookClusterDeploySuite) TearDownSuite() {
+	if mrc.T().Failed() {
+		gatherAllRookLogs(mrc.k8sh, mrc.Suite, defaultNamespace, clusterNamespace1)
+		gatherAllRookLogs(mrc.k8sh, mrc.Suite, defaultNamespace, clusterNamespace2)
+	}
+	deleteArgs := []string{"delete", "-f", "-"}
+
+	skipRookInstall := strings.EqualFold(mrc.installer.Env.SkipInstallRook, "true")
+	if skipRookInstall {
+		return
+	}
+	k8sVersion := mrc.k8sh.GetK8sServerVersion()
+	serverVersion, err := mrc.k8sh.Clientset.Discovery().ServerVersion()
+	if err != nil {
+		panic(err)
+	}
+	kubeVersion := utilversion.MustParseSemantic(serverVersion.GitVersion)
+
+	logger.Infof("Uninstalling All Rook Clusters")
+	k8sHelp, err := utils.CreateK8sHelper()
+	if err != nil {
+		panic(err)
+	}
+	rookOperator := mrc.installData.GetRookOperator(k8sVersion)
+
+	_, err = mrc.k8sh.KubectlWithStdin(rookOperator, deleteArgs...)
+	if err != nil {
+		logger.Errorf("Rook operator cannot be deleted,err -> %v", err)
+		panic(err)
+	}
+
+	_, err = k8sHelp.DeleteResource([]string{"-n", clusterNamespace1, "cluster", clusterNamespace1})
```

Comment:
again, there is a lot of duplication. I would think a helper could uninstall a namespace. Isn't that what `UninstallRookFromK8s` can do?

Reply:
To clarify, if I see two lines of code that are the same except for one of their parameters, it is code duplication. For example, these two lines are duplicated.
```
k8sHelp.DeleteResource([]string{"-n", clusterNamespace1, "cluster", clusterNamespace1})
k8sHelp.DeleteResource([]string{"-n", clusterNamespace2, "cluster", clusterNamespace2})
```
This pattern occurs a number of times. You could have a method something like this, and call it for each namespace. Thus, you avoid duplication and the code is easier to maintain.
```
CleanupCluster(namespace string){
  // error handling is missing
  k8sHelp.DeleteResource([]string{"-n", namespace, "cluster", namespace})
  k8sHelp.DeleteResource([]string{"-n", namespace, "serviceaccount", "rook-api"})
  ...
}
```